designer/client/src/components/modals/DeployWithParametersDialog.tsx

```diff
@@ -0,0 +1,137 @@
+import { css, cx } from "@emotion/css";
+import { WindowButtonProps, WindowContentProps } from "@touk/window-manager";
+import React, { useCallback, useMemo, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useDispatch, useSelector } from "react-redux";
+import { getActionParameters, getProcessName } from "../../reducers/selectors/graph";
+import { getFeatureSettings } from "../../reducers/selectors/settings";
+import { ProcessName } from "../Process/types";
+import { PromptContent, WindowKind } from "../../windowManager";
+import CommentInput from "../comment/CommentInput";
+import ProcessDialogWarnings from "./ProcessDialogWarnings";
+import { FormHelperText, Typography } from "@mui/material";
+import { LoadingButtonTypes } from "../../windowManager/LoadingButton";
+import { ActionNodeParameters } from "../../types/action";
+import { AdvancedParametersSection } from "./AdvancedParametersSection";
+import { mapValues } from "lodash";
+import { NodesDeploymentData } from "../../http/HttpService";
+import { ActionParameter } from "./ActionParameter";
+import { NodeTable } from "../graph/node-modal/NodeDetailsContent/NodeTable";
+
+export type ToggleProcessActionModalData = {
+    action: (processName: ProcessName, comment: string, nodeData: NodesDeploymentData) => Promise<unknown>;
+    displayWarnings?: boolean;
+};
+
+function initialNodesData(params: ActionNodeParameters[]) {
+    return params.reduce(
+        (paramObj, { nodeId, parameters }) => ({
+            ...paramObj,
+            [nodeId]: mapValues(parameters, (value) => value.defaultValue || ""),
+        }),
+        {},
+    );
+}
+
+export function DeployWithParametersDialog(props: WindowContentProps<WindowKind, ToggleProcessActionModalData>): JSX.Element {
+    // TODO: get rid of meta
+    const {
+        meta: { action, displayWarnings },
+    } = props.data;
+    const processName = useSelector(getProcessName);
+
+    const activityParameters = useSelector(getActionParameters);
+    const activityNodeParameters = activityParameters["DEPLOY"] || ([] as ActionNodeParameters[]);
+    const initialValues = useMemo(() => initialNodesData(activityNodeParameters), [activityNodeParameters]);
+    const [values, setValues] = useState(initialValues);
```

Comment:
Can we simplify it?

const [values, setValues] = useState(() => initialNodesData(activityNodeParameters));